src/cromshell/status/command.py

```diff
@@ -114,34 +115,74 @@ def main(config, workflow_id):
     return ret_val
 
 
-def get_metadata_status_summary(workflow_metadata):
+def get_all_call_values(nested_dictionary):
+    """Summarizes a given workflow metadata dictionary into a
+    new dictionary with only task name and status"""
+    temp_dic1 = {}
+    temp_dic2 = {}
+    temp_dic3 = {}
+    for key, value in nested_dictionary.items():
+        # Skip any keys that have values that are strings, because we are note
+        # interested in them
+        if type(value) is str:
+            continue
+        # If a dictionary value is encountered then we iterate through the dictionary
+        # to find call statuses
+        if type(value) is dict:
```

Comment:
It's worth having some comments here about using recursion for DFS in traversing the metadata tree.

Reply:
@SHuang-Broad Added this 
"# If a dictionary value is encountered then recursion is used iterate through
        # the dictionary to find call statuses. The reason for this is due the tree
        # structure of the metadata dictionary which holds the calls, subworkflow calls,
        # and their status within layers of the dictionary. Depth First Search recursion
        # is used to traverse the dictionary by iterating through any dictionary this 
        # this functions comes across first and obtaining their call and status. "